app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -75,10 +75,14 @@ class WifiDirectManager @Inject constructor(
   private var shouldRetry = true
 
   // Overall manager of Wifi p2p connections for the module
-  private lateinit var manager: WifiP2pManager
+  // Lazy initialize to counter Kt. TypeCastException on API 27;
```

Comment:
lazy isn't really solving anything here now that I take a more caffeinated look. Just changing the type to `WifiP2PManager?` is all that needs to be done. And of course casting the getService call to the nullable type with `as?` or `as WifiP2PManager?`

Reply:
To make initialization that shows `manager` as nullable, lazy is must; I can't think of doing so without adding extra lines of code

Reply:
`  private var manager: WifiP2pManager? = null`

Reply:
The problem is still same no, we gotta add one more line also for assigning it

Reply:
The problem isn't the amount of lines, the problem is that it is null on API 27. The minimal change to fix this is

```
private var manager: WifiP2pManager? = null
...
manager = activity.getSystemService(Context.WIFI_P2P_SERVICE) as? WifiP2pManager
```

and the best way is if it is a constructor parameter with a binding in a dagger module supplying it.
Lazy isn't wrong but it adds more noise, now we are thinking about ThreadSafetyMode and when this object gets instantiated.

Reply:
Why is it still null even if we're injecting it?